fix(docs-infra): make API member name bold

sjtrimble opened the pull request:
- ~~Remove linenums on API interface overview code block~~ _Addressed in a different PR._
- Return API member name bold font weight
- SCSS formatting cleanup

Closes #31494

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [x] Other... Please describe:

## What is the current behavior?
#31494

Issue Number: #31494

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

sjtrimble committed:
fix(docs-infra): make API member name bold

- Return API member name bold font weight
- SCSS formatting cleanup

Closes #31494

mary-poppins commented:
You can preview c938e43 at https://pr31574-c938e43.ngbuilds.io/.

mary-poppins commented:
You can preview 99df6be at https://pr31574-99df6be.ngbuilds.io/.

gkalpak commented on aio/src/styles/2-modules/_api-pages.scss:
```diff
@@ -80,15 +46,53 @@
+          margin: 16px 0;
+      }
+
+      tr:last-child {
```
Since, you are moving these around, it might make sense to merge this one with the `tr { ... }` rule set above:

```diff
 tr {
   border-bottom: 1px solid $lightgray;
+
+  &:last-child {
+    border-bottom: none;
+  }
 }
 ...
-tr:last-child {
-  border-bottom: none;
-}
```

sjtrimble replied:
Absolutely! Missed that one, thanks 👍

gkalpak commented on aio/src/styles/2-modules/_api-pages.scss:
```diff
@@ -80,15 +46,53 @@
     code-example {
         clear: left;
+      }
+    }
```
The indentation is really broken around here (and below), making it difficult to follow the rules.

gkalpak commented on aio/src/styles/2-modules/_api-pages.scss:
```diff
@@ -80,15 +46,53 @@
       }
+
+      th {
+          text-transform: none;
```
Since you are moving these around anyway, you could fix the indentation as well :tada:

sjtrimble commented on aio/src/styles/2-modules/_api-list.scss:
```diff
@@ -204,10 +204,6 @@ aio-api-list {
-
-      &.deprecated-api-item {
-        text-decoration: line-through;
-      }
```
It's duplicated code - it's already covered in the `api-pages.scss` file line 282.

gkalpak replied:
Yes, but there is `text-decoration: none` style [above][1], which has higher specificity than the one in `api-pages-scss`.
So, we need both.

[1]: https://github.com/angular/angular/blob/7ad9a62b1700dc68ca11ff7858a765200f4bc51d/aio/src/styles/2-modules/_api-list.scss#L198

gkalpak replied:
What about this comment? :smiley:

sjtrimble replied:
Ahhh I see. Ok, I shall return it!

sjtrimble commented on aio/tools/transforms/templates/api/includes/interface-overview.html:
```diff
@@ -1,10 +1,10 @@
 
 <section class="interface-overview">
-<code-example language="ts" hideCopy="true">
+<code-example language="ts" hideCopy="true" linenums="false">
```
I will only add it to `metadata.html` because the other two have a `no-box` class which I think gets rid of the lineums? Do you have an example of the other two in the page - it's hard to tell which ones use it.

gkalpak replied:
I don't think `.no-box` affects linenums 😕

`pipe-overview.html` is what you see in the overview of pipes (e.g. [SlicePipe][1]). Usually pipes have one-line overviews, that might change in the future. I think it makes sense to prevent linenums there (just in case). 

`annotations.html` is no longer used since #25677, so maybe we should remove it instead (see https://github.com/angular/angular/pull/25677#pullrequestreview-149625923 :grin:). But if it stays, I think it makes sense to prevent linenums for consistency.

[1]: https://next.angular.io/api/common/SlicePipe

gkalpak replied:
Actually...now that #31674 has been merged, we can remove all `linenums="false"` (since that is now the default) :tada:

sjtrimble replied:
Ok, I rebased and removed the `linenums="false"` reference

mary-poppins commented:
You can preview 7ad9a62 at https://pr31574-7ad9a62.ngbuilds.io/.

mary-poppins commented:
You can preview 169a80d at https://pr31574-169a80d.ngbuilds.io/.

gkalpak commented on aio/src/styles/2-modules/_api-list.scss:
```diff
@@ -205,7 +205,7 @@ aio-api-list {
       }
 
-      &.deprecated-api-item {
+       &.deprecated-api-item {
```
Incorrect indentation :grin:

gkalpak reviewed (approved):
One minor nit, plus that the last fixup commit (d6b76d160) does not match the original commit. It's header is `fixup! fix(docs-infra): return deprecated styles for api list` instead of `fixup! fix(docs-infra): remove lineums on api interface overview`.

@sjtrimble, once you've made the fixes, feel free to remove the `cleanup` label and add the `merge` label :wink:

sjtrimble commented:
@gkalpak Done, but I'd like to rebase too -- there are some conflicts I want to make sure are lining up.

ajitsinghkaler commented:
@gkalpak only cleanup is left should I clean up and make it ready for merge

mary-poppins commented:
You can preview dd5e430 at https://pr31574-dd5e430.ngbuilds.io/.

gkalpak committed:
fixup! fix(docs-infra): make API member name bold

gkalpak committed:
refactor(docs-infra): remove duplicate `.github-links` styles

mary-poppins commented:
You can preview 0491bf1 at https://pr31574-0491bf1.ngbuilds.io/.

gkalpak committed:
refactor(docs-infra): merge `.api-body` style blocks

This commit merges the two `.api-body` style blocks, removing duplicate
styles and re-ordering sub-blocks (to group relevant styles together).
It should not affect styling.

gkalpak commented:
@googlebot I consent.

gkalpak commented on aio/src/styles/2-modules/_api-pages.scss:
```diff
@@ -294,18 +294,6 @@
   }
 }
 
-.github-links {
```
There is a practically identical block further up this file.

gkalpak commented:
Thx for the ping, @ajitsinghkaler :+1:

I went ahead and cleaned it up myself, because you wouldn't have enough permissions to update the PR. While doing so, I realized that the linenums for interface overview had already been removed in an earlier PR, so this PR was mainly making API member name bold (and cleaning up SCSS code). I updated the commit messages to reflect that.

I also noticed that the `_api-pages.scss` code needed further clean-up (remove duplicate style and merge style-blocks). So, I pushed another two commits to address that. (My two commits should not have any impact on styling - just refactoring.)

@sjtrimble, @ajitsinghkaler: Please take a look at the changes/preview to make sure I didn't accidentally mess something up 🙏

mary-poppins commented:
You can preview af4aec9 at https://pr31574-af4aec9.ngbuilds.io/.

ajitsinghkaler commented:
@gkalpak most of it looks good just one question we have highlighted interface's classes elements
![Desktop screenshot (2)](https://user-images.githubusercontent.com/39260684/76138000-1fd35e80-6069-11ea-813f-c62ec4450fb7.png)

When I was checking out I saw type alias elements are not being highlighted any specific reason for that I think its best we highlight them too for consistency I 've attached screenshots for your reference. 

![Desktop screenshot (1)](https://user-images.githubusercontent.com/39260684/76137997-1649f680-6069-11ea-8de7-6fdd679f9a8b.png)


Rest LTGM.

ajitsinghkaler reviewed (commented):
Type-alias api type highlighting is still left

@gkalpak most of it looks good just one question we have highlighted interface's classes elements
![Desktop screenshot (2)](https://user-images.githubusercontent.com/39260684/76138000-1fd35e80-6069-11ea-813f-c62ec4450fb7.png)

When I was checking out I saw type alias elements are not being highlighted any specific reason for that I think its best we highlight them too for consistency I 've attached screenshots for your reference. 

![Desktop screenshot (1)](https://user-images.githubusercontent.com/39260684/76137997-1649f680-6069-11ea-8de7-6fdd679f9a8b.png)


Rest LGTM.

gkalpak commented:
Thx for taking a look, @ajitsinghkaler! I don't think we can fix this, because there doesn't seem to be a way to target `type`-alias members atm. For interfaces, they are identified as members and we can attach the `member-name` class to them, but `type`-aliases we don't have such granular tokenization.

I am not sure whether this is a limitation of TypeScript or whether it is something that can be fixed in dgeni, but in any case I believe this is outside of the scope of this PR.

Based on your feedback, I am marking this for merging and we can address any additional concerns/improvements in follow-up PRs.
